pkgs/development/tools/misc/sipp/default.nix

```diff
@@ -1,29 +1,30 @@
-{lib, stdenv, fetchurl, ncurses, libpcap }:
+{ lib, stdenv, fetchurl, ncurses, libpcap, cmake, openssl, git }:
 
 stdenv.mkDerivation rec {
-  version = "3.6.0";
+  version = "3.6.1";
 
   pname = "sipp";
 
   src = fetchurl {
-    url = "https://github.com/SIPp/${pname}/releases/download/v${version}/${pname}-${version}.tar.gz";
-    sha256 = "1fx1iy2n0m2kr91n1ii30frbscq375k3lqihdgvrqxn0zq8pnzp4";
+    url =
+      "https://github.com/SIPp/${pname}/releases/download/v${version}/${pname}-${version}.tar.gz";
+    sha256 = "sha256-alYOg6/5gvMx3byt+zvVMMWJbNW3V91utoITPMhg7LE=";
   };
 
-  postPatch = ''
-    sed -i "s@pcap/\(.*\).pcap@$out/share/pcap/\1.pcap@g" src/scenario.cpp
+  configurePhase = ''
+    cmake . -DUSE_GSL=1 -DUSE_PCAP=1 -DUSE_SSL=1 -DUSE_SCTP=0
   '';
 
-  configureFlags = [
-    "--with-pcap"
-  ];
+  buildPhase = ''
+    make -j4
+  '';
 
-  postInstall = ''
-    mkdir -pv $out/share/pcap
-    cp pcap/* $out/share/pcap
+  installPhase = ''
+    mkdir -p $out/bin
+    mv sipp $out/bin
   '';
 
-  buildInputs = [ncurses libpcap];
+  buildInputs = [ ncurses libpcap cmake openssl git ];
```

Comment:
We don't need cmake at runtime, so it goes into nativeBuildInputs instead. Same goes probably for git?

Reply:
Hmm, okay, maybe it's because of nix-update.
But when I omit cmake the build fails:
```
nix-update --shell sipp
$ nix eval --json --impure --experimental-features nix-command --expr (
    let
      inputs = (if (builtins.hasAttr "overlays" (builtins.functionArgs (import ./.))) then { overlays = []; } else { });
    in
    with import ./. inputs;
    let
      pkg = sipp;
      raw_version_position = builtins.unsafeGetAttrPos "version" pkg;

      position = if pkg ? isRubyGem then
        raw_version_position
      else
        builtins.unsafeGetAttrPos "src" pkg;
    in {
      name = pkg.name;
      old_version = (builtins.parseDrvName pkg.name).version;
      inherit raw_version_position;
      filename = position.file;
      line = position.line;
      urls = pkg.src.urls or null;
      url = pkg.src.url or null;
      rev = pkg.src.url.rev or null;
      hash = pkg.src.outputHash or null;
      mod_sha256 = pkg.modSha256 or null;
      vendor_sha256 = pkg.vendorSha256 or null;
      cargo_sha256 = pkg.cargoHash or pkg.cargoSha256 or null;
      tests = builtins.attrNames (pkg.passthru.tests or {});
    })
fetch https://github.com/SIPp/sipp/releases.atom
Not updating version, already 3.6.1
$ nix-shell /var/folders/vz/xyrkgnq92mx38v189hk6swbm0000gq/T/tmpy4ya3ypz/default.nix
this derivation will be built:
  /nix/store/90i5gbca96nnxd6s4c58vk1hra6ff0xn-sipp-3.6.1.drv
building '/nix/store/90i5gbca96nnxd6s4c58vk1hra6ff0xn-sipp-3.6.1.drv'...
unpacking sources
unpacking source archive /nix/store/izq8s473yngcspikgafzcx2ppk8jp5bw-sipp-3.6.1.tar.gz
source root is sipp-3.6.1
setting SOURCE_DATE_EPOCH to timestamp 1600287963 of file sipp-3.6.1/sipp.1
patching sources
configuring
/nix/store/1ym02gx4cmvrxfr1drxfg1dqs9djla86-stdenv-darwin/setup: line 1362: cmake: command not found
error: builder for '/nix/store/90i5gbca96nnxd6s4c58vk1hra6ff0xn-sipp-3.6.1.drv' failed with exit code 127;
       last 7 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/izq8s473yngcspikgafzcx2ppk8jp5bw-sipp-3.6.1.tar.gz
       > source root is sipp-3.6.1
       > setting SOURCE_DATE_EPOCH to timestamp 1600287963 of file sipp-3.6.1/sipp.1
       > patching sources
       > configuring
       > /nix/store/1ym02gx4cmvrxfr1drxfg1dqs9djla86-stdenv-darwin/setup: line 1362: cmake: command not found

```

Regarding git, cmake gives me an info that it can't find git

```
-- Looking for le16toh - not found
-- Could NOT find Git (missing: GIT_EXECUTABLE)
-- Configuring done
```
However, that doesn't seem to affect the build.

Reply:
You're not supposed to omit cmake, but instead put it into `nativeBuildInputs` as opposed to `buildInputs`.

Reply:
Thanks. Saw your other comments after that one. Makes sense now.